albumentations/augmentations/transforms.py

```diff
@@ -3082,3 +3083,44 @@ def apply_to_mask(self, img, dropout_mask=None, **params):
 
     def get_transform_init_args_names(self):
         return ("max_objects", "image_fill_value", "mask_fill_value")
+
+
+class ShotNoise(ImageOnlyTransform):
+    """Apply shot   noise to the input image.
+    Args:
+        noise_severity (int): severity of added noise. Lower values mean higher distortion.
+            Should be in range [3, inf). Default: (12).
+        p (float): probability of applying the transform. Default: 0.5.
+    Targets:
+        image
+    Image types:
+        uint8, float32
+
+    Reference:
+    |  https://arxiv.org/abs/1903.12261
+    |  https://github.com/hendrycks/robustness/blob/master/ImageNet-C/create_c/make_imagenet_c.py
+    """
+
+    def __init__(self, noise_severity=12, always_apply=False, p=0.5):
+        super(ShotNoise, self).__init__(always_apply, p)
+
+        assert noise_severity >= 3
+
+        self.noise_severity = noise_severity
+
+    def apply(self, img, shotnoise=None, **params):
+        return F.shot_noise(img, shotnoise=shotnoise)
+
+    def get_params_dependent_on_targets(self, params):
+        image = params["image"]
+        image = np.array(image) / 255.0
```

Comment:
How will this transform work for non-uint8 images? Should it be another factor here that depends on the dtype?

Reply:
Fixed it